Rubberduck.Parsing/ComReflection/ComProject.cs

```diff
@@ -100,10 +101,10 @@ private void LoadModules(ITypeLib typeLibrary)
                     {
                         var typeAttributes = Marshal.PtrToStructure<TYPEATTR>(typeAttributesPointer);
                         KnownTypes.TryGetValue(typeAttributes.guid, out var type);
-
-                        switch (typeAttributes.typekind)
+                        
+                        switch (typeAttributes.typekind.ToTypeKindVbe())
```

Comment:
I think that if we're going to assign `TKIND_VBACLASS`, we should at least make sure it's actually a VBA class at least.

Reply:
How would you do that?

Moreover, nothing on a usual typelib should ever have the end of enum marker at TYPEKIND.

Reply:
I originally thought of using the `HasVBEExtensions` function but looking at it, it might make more sense to simply set up the `ComProject` so that if it receives a VBE's `ITypeLib` from the TLAPI, set a property as a guard against mislabeling a non-VBA type info as VBA. 

Alternatively, Wayne suggested simply casting it into `TKIND_INTERFACE`, since only the TLAPI cares about that distinction, but not the COM collector, right?

Reply:
@bclothier actually, I think this does need separate handling, since a TKIND_INTERFACE would not be expected to have any fields (cVars should be 0).

Reply:
If you want to double-check the TKIND_VBACLASS value actually came from the type lib API, you could just test if the ITypeInfo casts to a TypeInfoWrapper.